packages/rest/src/body-parsers/helper.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as debugModule from 'debug';
+import {HttpError} from 'http-errors';
+import {Request, Response, RequestBodyParserOptions} from '../types';
+
+import {OptionsJson, OptionsUrlencoded, OptionsText} from 'body-parser';
+const debug = debugModule('loopback:rest:body-parser');
+
+/**
+ * Get the content-type header value from the request
+ * @param req Http request
+ */
+export function getContentType(req: Request): string | undefined {
+  return req.get('content-type');
+}
+
+/**
+ * Express body parser function type
+ */
+export type BodyParserWithCallback = (
+  request: Request,
+  response: Response,
+  callback: (err: HttpError) => void,
+) => void;
+
+/**
+ * Normalize parsing errors as `4xx`
+ * @param err
+ */
+export function normalizeParsingError(err: HttpError) {
```

Comment:
As I mentioned in the other comment, I think error normalization should be handled in a single place and that's in `RequestBodyParser`. Let's move this helper function to `body-parser.ts`.